### PR TITLE
feat(progress-tracker): add base markup for progress-tracker

### PR DESCRIPTION
**Pull request by soykje**
**TASK**: #1581 

### Description, Motivation and Context
Getting started with component, with base features.

**NB:** Interactivity behaviour will be updated/refactored after specs updates, in further PR.

### Types of changes
- [x] ✨ New feature (non-breaking change which adds functionality)
- [x] 🧾 Documentation
- [x] 🧪 Test
- [x] 💄 Styles

### CODE REVIEW

File: packages/components/progress-tracker/src/ProgressTrackerStep.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    const stepId = `step-${ID}`
+    const stepIndex = [...steps.keys()].indexOf(stepId)
+
+    const disabledAfter = useMemo(() => {
+      const nextStepId = [...steps.keys()][stepIndex + 1]
+
+      return !!(nextStepId && steps.get(nextStepId)?.includes('disabled'))
+    }, [steps, stepIndex])
+
+    const [progressState, setProgressState] =
```
Comment (acd02) [inline packages/components/progress-tracker/src/ProgressTrackerStep.tsx:52]: I think we can do without this `useState`, i.e.:
```tsx
progressState = (() => {
  if (stepIndex === activeStepIndex) 'active'
  else if (stepIndex < activeStepIndex) return 'complete'
  return 'incomplete'
})()
```

Comment (soykje) [reply packages/components/progress-tracker/src/ProgressTrackerStep.tsx:52]: Thx, I'll test this option :+1: But to be curious, what would be the real gain of this method?

Comment (acd02) [reply packages/components/progress-tracker/src/ProgressTrackerStep.tsx:52]: https://react.dev/learn/you-might-not-need-an-effect#chains-of-computations

I would flip this question around, and ask you what's the benefit of using `useState` and `useEffect` when you can get by without them 😉

Comment (soykje) [reply packages/components/progress-tracker/src/ProgressTrackerStep.tsx:52]: I knew you would send this link...

File: packages/components/progress-tracker/src/ProgressTracker.tsx [hunk 2]
```diff
@@ -1,9 +1,84 @@
...
+  /**
+   * The size of the step indicators
+   * @default 'lg'
+   */
+  size?: 'sm' | 'md' | 'lg'
+  /**
+   * The index of the current step.
+   * @default 0
+   */
+  stepIndex?: number
```
[the remark this answers is not in the document]
Comment (soykje) [reply packages/components/progress-tracker/src/ProgressTracker.tsx:25]: I don't understand why we should change now, as it has been previously discussed without any remarks on this naming :thinking:
Anyway, I think we should keep the `*Index` information, as prop expects... an index :sweat_smile: Just took a look on some other DS like [Chakra](https://chakra-ui.com/docs/components/stepper/props#props) and not sure the `active` or `current` is more relevant here... But I don't have a really strong opinion on it^^ Happy to discuss.

File: packages/components/progress-tracker/src/ProgressTrackerStep.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      setSteps,
+      size,
+      readOnly,
+    } = useProgressTrackerContext()
+
+    const ID = useId()
+    const stepId = `step-${ID}`
+    const stepIndex = [...steps.keys()].indexOf(stepId)
+
+    const disabledAfter = useMemo(() => {
```
[the remark this answers is not in the document]
Comment (soykje) [reply packages/components/progress-tracker/src/ProgressTrackerStep.tsx:41]: You're right, as deps come from props/context, reconciliation already works :+1:

**Review by Powerplex — APPROVED** [review]

**Review by acd02 — APPROVED** [review]

**Comment by andresin87** [comment]
It looks like there is a problem with the interaction on this PR

https://github.com/adevinta/spark/assets/1674036/c7eea564-2cae-448c-a2b6-e9aaeeb5d480

We need to review at the end of all the PRs

**Comment by andresin87** [comment]
progress-tracker is using basic, main, support naming convention and this is only availabe using
"@spark-ui/theme-utils": "^4.0.0",
as a peer-dependency. Please do not forget

**Review by andresin87 — APPROVED** [review]

**Pull request merged**